--- openssl.spec	2014-08-15 18:44:02.000000000 -0300
+++ openssl.fedora	2014-08-15 18:46:18.000000000 -0300
@@ -28,9 +28,8 @@ Epoch: 1
 # We have to remove certain patented algorithms from the openssl source
 # tarball with the hobble-openssl script which is included below.
 # The original openssl upstream tarball cannot be shipped in the .src.rpm.
-#Source: openssl-%{version}-hobbled.tar.xz
-Source: openssl-%{version}.tar.gz
-#Source1: hobble-openssl
+Source: openssl-%{version}-hobbled.tar.xz
+Source1: hobble-openssl
 Source2: Makefile.certificate
 Source6: make-dummy-cert
 Source7: renew-dummy-cert
@@ -89,9 +88,6 @@ Patch81: openssl-1.0.1-beta2-padlock64.p
 Patch84: openssl-1.0.1i-trusted-first.patch
 Patch85: openssl-1.0.1e-arm-use-elf-auxv-caps.patch
 Patch89: openssl-1.0.1e-ephemeral-key-size.patch
-# Fix for EL < 7.
-# The release 300 is to avoid conflicting with future patches.
-Patch300: openssl-1.0.1i-fix_secure_gentenv.patch
 
 License: OpenSSL
 Group: System Environment/Libraries
@@ -162,7 +158,7 @@ from other formats to the formats used b
 
 # The hobble_openssl is called here redundantly, just to be sure.
 # The tarball has already the sources removed.
-#%{SOURCE1} > /dev/null
+%{SOURCE1} > /dev/null
 
 cp %{SOURCE12} %{SOURCE13} crypto/ec/
 
@@ -213,7 +209,6 @@ cp %{SOURCE12} %{SOURCE13} crypto/ec/
 %patch84 -p1 -b .trusted-first
 %patch85 -p1 -b .armcap
 %patch89 -p1 -b .ephemeral
-%patch300 -p1 -b .fix_secure_gentenv
 
 sed -i 's/SHLIB_VERSION_NUMBER "1.0.0"/SHLIB_VERSION_NUMBER "%{version}"/' crypto/opensslv.h
 
@@ -268,12 +263,11 @@ sslarch="linux-ppc64le"
 # Configure the build tree.  Override OpenSSL defaults with known-good defaults
 # usable on all platforms.  The Configure script already knows to use -fPIC and
 # RPM_OPT_FLAGS, so we can skip specifiying them here.
-# Note: enabling ec2m WILL require you already have openssl-fips in your system
 ./Configure \
 	--prefix=%{_prefix} --openssldir=%{_sysconfdir}/pki/tls ${sslflags} \
 	--system-ciphers-file=%{_sysconfdir}/crypto-policies/back-ends/openssl.config \
 	zlib enable-camellia enable-seed enable-tlsext enable-rfc3779 \
-	enable-cms enable-md2 no-mdc2 no-rc5 no-ec2m enable-ec enable-ecdh enable-ecdsa enable-srp \
+	enable-cms enable-md2 no-mdc2 no-rc5 no-ec2m no-gost no-srp \
 	--with-krb5-flavor=MIT --enginesdir=%{_libdir}/openssl/engines \
 	--with-krb5-dir=/usr shared  ${sslarch} %{?!nofips:fips}
 
